sample-apps/client/ts-quickstart/src/closed-captions.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  Call,
+  CallClosedCaption,
+  ClosedCaptionEvent,
+} from '@stream-io/video-client';
+
+export class ClosedCaptionManager {
+  status: 'on' | 'off' = 'off';
+  private unsubscribe?: () => void;
+  private captionRefreshInterval?: ReturnType<typeof setInterval>;
+  private captions: (CallClosedCaption & { speaker_name?: string })[] = [];
+  private captionContainer?: HTMLElement;
+
+  constructor(private call: Call) {}
+
+  renderToggleElement() {
+    const button = document.createElement('button');
+    button.textContent =
+      this.status === 'on'
+        ? 'Turn off closed captions'
+        : 'Turn on closed captions';
+
+    button.addEventListener('click', async () => {
+      this.status === 'on' ? this.hideCaptions() : this.showCaptions();
+      button.textContent =
+        this.status === 'on'
+          ? 'Turn off closed captions'
+          : 'Turn on closed captions';
+    });
+
+    return button;
+  }
+
+  renderCaptionContainer() {
+    this.captionContainer = document.createElement('div');
+
+    return this.captionContainer;
+  }
+
+  showCaptions() {
+    this.status = 'on';
+    this.unsubscribe = this.call.on(
+      'call.closed_caption',
+      (event: ClosedCaptionEvent) => {
+        const caption = event.closed_caption;
+        const isDuplicate = this.captions.find(
+          (c) =>
+            c.speaker_id === caption.speaker_id &&
+            c.start_time === caption.start_time,
+        );
+        if (!isDuplicate) {
+          const speaker = this.call.state.participants.find(
+            (p) => p.userId === caption.speaker_id,
+          );
+          const speakerName = speaker?.name || speaker?.userId;
+          this.captions.push({ ...caption, speaker_name: speakerName });
+          if (this.captions.length === 1) {
+            this.updateDisplayedCaptions();
+          }
+        }
+      },
+    );
+    this.captionRefreshInterval = setInterval(() => {
+      this.captions = this.captions.slice(1);
+      this.updateDisplayedCaptions();
+    }, 2700);
```

Comment:
And this timeout as well :)

Reply:
Implemented in https://github.com/GetStream/stream-video-js/pull/1497/commits/acc06670c736c32d1ccef53da6c1ff7c53a3673c